### PR TITLE
Cleanup LowerBlockStore

### LINKED ISSUE #27169: initblk incorrectly handles 0 fill on ARM64
The fill values passed to initblk is supposed to be a byte but ARM64 block store lowering fails to mask the upper bytes resulting in incorrect initialization:
```
ldloca V_0
ldc.i4 0x11100
ldc.i4 15
initblk
```
generates:
```asm
52822000          movz    w0, #0x1100
72A00020          movk    w0, #1 LSL #16
9101C3A1          add     x1, fp, #112  // [V05 loc0]
F9000020          str     x0, [x1]     ; stores 0x0000000000011100
B9000820          str     w0, [x1,#8]  ; stores 0x00011100
79001820          strh    w0, [x1,#12] ; stores 0x1100
39003820          strb    w0, [x1,#14] ; stores 0x00
```

https://github.com/dotnet/coreclr/pull/27035#discussion_r331629601

### PR DESCRIPTION
**Pull request by mikedn**
Extracted from #27035

Fixes #27169

### CODE REVIEW

**Comment by mikedn** [comment]
@sandreenko Is it possible to re-run only "Run Test Pri0 Linux arm64 checked"? I would like to check that the test I added catches the bug before adding the actual fix.

**Comment by sandreenko** [comment]
> @sandreenko Is it possible to re-run only "Run Test Pri0 Linux arm64 checked"? I would like to check that the test I added catches the bug before adding the actual fix.

I think we don't have that precision(and probably don't need), so I have restarted all failed checks.

Should not `Run Test Pri0 Windows_NT arm64 checked` repro that as well?

**Comment by mikedn** [comment]
> Should not Run Test Pri0 Windows_NT arm64 checked repro that as well?

Sure, if that does anything. I looked at a successful build and noticed that the "Send tests to Helix" step is actually skipped in that job so I'm not sure what that job does, if anything.

Thanks!

**Comment by mikedn** [comment]
Cool, test failed: https://dev.azure.com/dnceng/public/_build/results?buildId=386645&view=ms.vss-test-web.build-test-results-tab&runId=11977646&resultId=100695&paneView=debug

File: src/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -280,93 +253,69 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
...
-            unsigned slots = objNode->GetLayout()->GetSlotCount();
-
-#ifdef DEBUG
-            // CpObj must always have at least one GC-Pointer as a member.
-            assert(objNode->GetLayout()->HasGCPtr());
-
-            assert(dstAddr->gtType == TYP_BYREF || dstAddr->gtType == TYP_I_IMPL);
+            // TODO-Cleanup: Make sure that GT_IND lowering didn't mark the source address as contained.
+            // Sometimes the GT_IND type is a non-struct type and then GT_IND lowering may contain the
+            // address, not knowing that GT_IND is part of a block op that has containment restrictions.
```
Comment (CarolEidt) [inline src/jit/lowerarmarch.cpp]: Thanks for adding this clarifying comment!

**Pull request merged**